src/track/track_gps_l2cm.c

```diff
@@ -0,0 +1,614 @@
+                   L2CM_TRACK_SHORT_CYCLE_INTERVAL_CHIPS);
+    break;
+
+  default:
+    assert(0);
+    break;
+  }
+}
+
+static void tracker_gps_l2cm_update(const tracker_channel_info_t *channel_info,
+                                    tracker_common_data_t *common_data,
+                                    tracker_data_t *tracker_data)
+{
+  gps_l2cm_tracker_data_t *data = tracker_data;
+
+  if (data->startup) {
+    /* Handle start-up case, when we have 1+1+18 integration cycles */
+    handle_tracker_startup(channel_info, common_data, data);
+    if (1 == data->startup) {
+      // The start-up phase is over. Continue with the normal
+      // sequence of short & long cycles. The next cycle
+      // is going to be long.
+      data->short_cycle = false; // start-up phase is over
+    }
+    if (data->startup) {
+      data->startup--;
+    }
+    return;
+  }
+
+  /* Read early ([0]), prompt ([1]) and late ([2]) correlations. */
+  if (data->short_cycle) {
+    tracker_correlations_read(channel_info->context, data->cs,
+                              &common_data->sample_count,
+                              &common_data->code_phase_early,
+                              &common_data->carrier_phase);
+    alias_detect_first(&data->alias_detect, data->cs[1].I, data->cs[1].Q);
+  } else {
+    /* This is the end of the long cycle's correlations. */
+    corr_t cs[3];
+    tracker_correlations_read(channel_info->context, cs,
+                              &common_data->sample_count,
+                              &common_data->code_phase_early,
+                              &common_data->carrier_phase);
+    /* Accumulate short cycle correlations with long ones. */
+    for(int i = 0; i < 3; i++) {
+      data->cs[i].I += cs[i].I;
+      data->cs[i].Q += cs[i].Q;
+    }
+  }
+
+  u8 int_ms = data->short_cycle ?
+              L2CM_TRACK_SHORT_CYCLE_INTERVAL_MS :
+              L2CM_TRACK_LONG_CYCLE_INTERVAL_MS;
+  common_data->TOW_ms = tracker_tow_update(channel_info->context,
+                                           common_data->TOW_ms,
+                                           int_ms);
+
+  /* We're doing long integrations, alternate between short and long
+   * cycles. This is because of FPGA pipelining and latency.
+   * The loop parameters can only be updated at the end of the second
+   * integration interval.
+   */
+  bool short_cycle = data->short_cycle;
+
+  data->short_cycle = !data->short_cycle;
+
+  if (short_cycle) {
+    tracker_retune(channel_info->context, common_data->carrier_freq,
+                   common_data->code_phase_rate,
+                   L2CM_TRACK_SHORT_CYCLE_INTERVAL_CHIPS);
+    return;
+  }
+
+  common_data->update_count += data->int_ms;
+
+  /* L2C bit sync is known once we start tracking it since
+     the L2C ranging code length matches the bit length (20ms).
+     This is the end of 20ms integration period and the edge
+     of a data bit. */
+  tracker_bit_sync_set(channel_info->context, 0);
```

Comment:
No, it is not reasonable. You are right. I moved it to the tracker_gps_l2cm_init().